src/CloudNative.CloudEvents.Amqp/AmqpClientExtensions.cs

```diff
@@ -94,5 +95,96 @@ private static bool HasCloudEventsContentType(Message message, out string conten
             contentType = (message.Properties.ContentType as Symbol)?.ToString();
             return contentType?.StartsWith(CloudEvent.MediaType) == true;
         }
+
+        public static Message ToAmqpMessage(this CloudEvent cloudEvent, ContentMode contentMode, CloudEventFormatter formatter)
+        {
+            var applicationProperties = MapHeaders(cloudEvent);
+            RestrictedDescribed bodySection;
+            Properties properties;
+
+            switch (contentMode)
+            {
+                case ContentMode.Structured:
+                    bodySection = new Data
+                    {
+                        Binary = formatter.EncodeStructuredModeMessage(cloudEvent, out var contentType)
+                    };
+                    // TODO: What about the other parts of the content type?
+                    properties = new Properties { ContentType = contentType.MediaType };
```

Comment:
Hmm, the spec doesn't indicate that ContentType should be set for structured mode - https://github.com/cloudevents/spec/blob/v1.0/amqp-protocol-binding.md#13-content-modes

Reply:
It's specified here: https://github.com/cloudevents/spec/blob/v1.0/amqp-protocol-binding.md#321-amqp-content-type

(The doc isn't very clear, admittedly...)